File: backend/springnatour/src/main/java/com/unina/springnatour/dto/compilation/CompilationDto.java
package com.unina.springnatour.dto.compilation;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.unina.springnatour.dto.route.RouteDto;
import com.unina.springnatour.dto.user.UserDto;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class CompilationDto {
    @JsonProperty("compilation_id")
    private Long compilationId;

    @JsonProperty("compilation_title")
    private String compilationTitle;

    @JsonProperty("compilation_description")
    private String compilationDescription;

    @JsonProperty("compilation_creation_date")
    private LocalDateTime compilationCreationDate;

    @JsonProperty("compilation_photo")
    private String compilationPhoto;

    @JsonProperty("compilation_author")
    private UserDto author;
}

File: client/app/src/main/java/com/unina/natourkt/data/remote/dto/CompilationDto.kt
package com.unina.natourkt.data.remote.dto

import com.google.gson.annotations.SerializedName
import com.unina.natourkt.domain.model.Compilation
import java.time.LocalDateTime

data class CompilationDto(
    @SerializedName("compilation_id")
    val id: Long,

    @SerializedName("compilation_title")
    val title: String,

    @SerializedName("compilation_description")
    val description: String,

    @SerializedName("compilation_creation_date")
    val creationDate: String,

    @SerializedName("compilation_photo")
    val photo: String,

    @SerializedName("compilation_author")
    val author: UserDto
)